gen3cirrus/aws/utils.py

```diff
@@ -0,0 +1,130 @@
+import datetime
+import json
+
+import boto3
+from botocore.exceptions import ClientError
+
+from cdislogging import get_logger
+
+logger = get_logger(__name__, log_level="info")
+
+
+def generatePresignedURL(
+    client, method, bucket_name, object_name, expires, additional_info={}
```

Comment:
It's not safe to default a mutable type (like a dict) as a default value in function definitions. If you do it like this, the same mutable object gets used *every time the function is called* without explicitly providing that argument, which is (in almost every case) not the behavior you really intend b/c any changes to the variable get *persisted* between function calls.

In this case, I don't think you actually modify it, but it's considered a dangerous practice.

Check this out if you want to see this in action, the result is fun:

```python
def add_item_to_dict(item, my_dict={}):
    """Function to add an item to a dictionary."""
    if item in my_dict:
        my_dict[item] += 1
    else:
        my_dict[item] = 1
    return my_dict

# Call the function multiple times
print("First call:", add_item_to_dict('apple'))
print("Second call:", add_item_to_dict('banana'))
print("Third call:", add_item_to_dict('apple'))
```

The proper way to work around this pitfall is to use `None` and then set the mutable default in the actual function. `additional_info=None` in the definition, then in the function: `additional_info = additional_info or {}`

Reply:
good suggestion! refactored